File: relationship-go/modedata/mode.go
/*
模式数据
*/

package modedata

import (
	"maps"

	"github.com/young2j/relationship/relationship-go/datamap"
	"github.com/young2j/relationship/relationship-go/locale"
	"github.com/young2j/relationship/relationship-go/options"
)

// 获取模式数据
func GetModeData(mode options.OptionMode) map[string][]string {
	data := maps.Clone(datamap.InitMap())

	if mode == options.OptionModeGuangDong {
		for k, v := range locale.Guangdong.Data {
			if dat, ok := data[k]; ok {
				data[k] = append(v, dat...)
			} else {
				data[k] = v
			}
		}

	} else if mode == options.OptionModeNorth {
		for k, v := range locale.North.Data {
			if dat, ok := data[k]; ok {
				data[k] = append(v, dat...)
			} else {
				data[k] = v
			}
		}
	}

	return data
}

File: relationship-py/relationship/mode.py
"""
模式数据
"""

from .options import OptionMode
from .map import init_map


# 获取模式数据
def get_mode_data(mode: OptionMode):
    data = {**init_map()}
    if mode == OptionMode.GUANG_DONG:
        from relationship.locale_ import guangdong

        for k, v in guangdong["data"].items():
            data[k] = v + data.get(k, [])

    elif mode == OptionMode.NORTH:
        from  relationship.locale_ import north

        for k, v in north["data"].items():
            data[k] = v + data.get(k, [])

    return data
